micropip/install.py

```diff
@@ -60,16 +63,20 @@ async def install(
 
         pyodide_packages, wheels = transaction.pyodide_packages, transaction.wheels
 
-        package_names = [pkg.name for pkg in wheels + pyodide_packages]
+        packages_all = [pkg.name for pkg in wheels + pyodide_packages]
+        distributions = search_installed_packages(packages_all)
+
+        with indent_log():
+            uninstall_distributions(distributions, logger)
 
```

Comment:
Shouldn't this only happen if `reinstall` is `True`?

Reply:
The `distributions` parameter will always be an empty list if reinstall is false, as the transaction will not add any package that are already installed when reinstall is false.

But I think I can add that check just in case.